rpc_util.go

```diff
@@ -622,7 +622,7 @@ func (pf payloadFormat) isCompressed() bool {
 }
 
 type streamReader interface {
-	ReadHeader(header []byte) error
+	ReadMessageHeader(header []byte) error
 	Read(n int) (mem.BufferSlice, error)
```

Comment:
From the description:
> (I found `ReadHeader` ambiguous and confusing) 

We have other headers, and when I read `ReadHeaders` my mind immediately jumped to that metadata kind of header, not the message header.  I'm fine with `ReadMessage`.  But I think I'd like to go further and delete `ReadMessageHeader` entirely and move the functionality of "getting the next message" into the stream API as a first-class concept.

Reply:
Actually, I will keep `Read` unless you feel strongly (but you sound like you aren't) since it actually just reads raw bytes, and could read the header bytes just as easily.

Reply:
Sounds good.